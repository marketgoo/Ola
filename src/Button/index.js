import React from 'react'
import { default as PT } from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../utils'

import Spinner from '../Spinner'
import Icon, { ICONS } from '../Icon'

const Button = ({
  variant,
  children,
  disabled,
  multiline,
  busy,
  className,
  icon,
  as,
  selected,
  progress,
  ...props
}) => {
  const ElementType = getElementType(Button, { as: as, ...props })
  delete props['as']
  const styles = cx(
    'ola_button',
    { [`is-${variant}`]: variant },
    { 'is-busy': busy },
    { 'is-multiline': multiline },
    { 'is-disabled': disabled && !busy },
    className
  )

  const pressed = selected === null ? null : selected ? 'true' : 'false'

  const iconContent =
    typeof icon === 'string' ? (
      <Icon name={icon} size="small" className="ola_button-icon" />
    ) : (
      <span className="ola_icon ola_button-icon is-small">{icon}</span>
    )

  return (
    <ElementType
      className={styles}
      disabled={busy ? true : disabled}
      aria-pressed={pressed}
      {...props}
    >
      {busy ? <Spinner /> : icon && iconContent}
      <span className="ola_button-text">{!busy ? children : busy}</span>
      {typeof progress === 'number' ? (
        <span className="ola_button-progress">{clamp(progress)}%</span>
      ) : null}
    </ElementType>
  )
}

Button.defaultProps = {
  as: 'button',
  variant: null,
  busy: null,
  className: null,
  disabled: false,
  multiline: null,
  selected: null,
  progress: null,
  icon: null,
}

Button.propTypes = {
  /** Render Button with any html tag */
  as: PT.string,
  /** Button variants */
  variant: PT.oneOf([
    'primary',
    'secondary',
    'destructive-primary',
    'destructive',
    'pro',
    'pro-secondary',
    'link',
  ]),
  /** Text for loading state */
  busy: PT.oneOfType([PT.string, PT.bool]),
  /** Extra className */
  className: PT.string,
  /** Button content */
  children: PT.oneOfType([
    PT.oneOf(Object.keys(ICONS)),
    PT.arrayOf(PT.node),
    PT.node,
  ]).isRequired,
  /** Whether the button is disabled */
  disabled: PT.bool,
  /** Whether the button include multiline text */
  multiline: PT.bool,
  /** Whether the button is selected */
  selected: PT.bool,
  /** The percentage progress */
  progress: PT.number,
  /** Icon name  */
  icon: PT.oneOfType([PT.string, PT.arrayOf(PT.node), PT.node]),
}

export default Button

function clamp(number) {
  return Math.min(Math.max(number, 0), 100)
}
