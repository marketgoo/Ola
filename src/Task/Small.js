import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../utils'
import Icon from '../Icon'

const TaskIcon = ({ variant }) => {
  switch (variant) {
  case 'success': return (<Icon name='success' size="small" className='ola_task-status' />)
  case 'warning': return (<Icon name='warning' size="small" className='ola_task-status' />)
  case 'suggested': return (<Icon name='help' size="small" className='ola_task-status' />)
  default: return (<Icon name='error' size="small" className='ola_task-status' />)
  }
}

const TaskSmall = ({ variant, children, as, className, ...props }) => {
  const ElementType = getElementType(TaskSmall, { as: as, ...props })

  return (
    <ElementType className={cx('ola_task', 'is-small', className, variant && `is-${variant}`)} {...props}>
      <p className="ola_task-header">
        <TaskIcon variant={variant} />
        <span className="ola_task-title">{children}</span>
      </p>
    </ElementType>
  )
}

TaskSmall.defaultProps = {
  as: 'div',
  variant: 'error',
  className: null,
}

TaskSmall.propTypes = {
  /** Render Task with any html tag */
  as: PT.string,
  variant: PT.oneOf(['success', 'error', 'suggested', 'warning']),
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default TaskSmall
