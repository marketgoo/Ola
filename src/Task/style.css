.ola_task {
    --inner: var(--size-8);

    margin: 0;
    background: var(--white);

    &.is-success {
        & .ola_task-status {
            fill: var(--success);
        }
    }
    &.is-error {
        & .ola_task-status {
            fill: var(--error);
        }
    }
    &.is-warning {
        & .ola_task-status {
            fill: var(--warning);
        }
    }
    &.is-suggested {
        & .ola_task-status {
            fill: var(--gray);
        }
    }

    & .ola_task-title {
            flex: 1 1 auto;
            @media (--to-width-1) {
                text-overflow: ellipsis;
                overflow: hidden;
                white-space: nowrap;
            }
    }

    & .ola_task-extra {
        margin-left: var(--size-6);
    }

    &[open] {
        --inner: calc(var(--size-8) + var(--size-4));

        position: relative;
        z-index: 2;
        box-shadow: var(--shadow-2), var(--shadow-line);
        margin-left: calc(var(--size-4) * -1);
        margin-right: calc(var(--size-4) * -1);
        border-radius: var(--radius);
        padding-top: var(--size-4);
        margin-top: calc(var(--size-4) * -1);
        margin-top: var(--size-4);

        & .ola_task-content {
            animation: task-open 200ms;
            animation-fill-mode: both;
            margin-left: var(--size-8);
        }

        & .ola_task-icon {
            display: block;
        }

        & .ola_task-summary {
            color: inherit;

            &:focus-visible {
                box-shadow: none;
            }
            &:hover {
                background: none;
            }
        }


        & .ola_task-extra {
            display: none;
        }

        & .ola_task-title {
            @media (--to-width-1) {
                text-overflow: initial;
                overflow: auto;
                white-space: initial;
                padding-right: 0;
            }
        }

        @media (--to-width-2) {
            margin-left: 0;
            margin-right: 0;
        }
    }
}
.ola_task.is-small {
    & .ola_task-header {
        font-size: var(--font-callout);
        min-height: var(--size-8);
        margin: 0;
        padding: 0;
    }
    & .ola_task-status {
        margin-right: var(--size-3);
    }
}
a.ola_task.is-small {
    & .ola_task-header {
        cursor: pointer;
    }
}

.ola_task-header {
    font: var(--font-body);
    list-style: none;
    display: flex;
    align-items: center;
    padding: 0 var(--inner);
    cursor: default;
    position: relative;
    z-index: 1;
    min-height: 52px;

    @media (--to-width-2) {
        padding: 0 var(--size-6);
    }
}

.ola_task-summary {
    color: var(--black);
    cursor: pointer;
    transition: background-color var(--duration);
    outline: 0;
    list-style: none;

    &::-webkit-details-marker {
        display: none;
    }

    &:hover {
        color: var(--brand-dark);
        background-color: var(--gray-xlight);
    }

    &:focus {
        position: relative;
        z-index: 2;
        box-shadow: 0 0 0 4px var(--brand-focus);
    }
}
.ola_task-status {
    flex: 0 0 auto;
    margin-right: var(--size-5);
}
.ola_task-icon {
    flex: 0 0 auto;
    margin-left: auto;
    margin-right: calc(var(--size-7) * -1);
    display: none;
    color: var(--gray);

    @media (--to-width-2) {
        margin-right: 0;
    }
}
.ola_task-body {
    padding: var(--size-4) var(--inner) var(--size-8);
    font: var(--font-callout);
    color: var(--gray);

    & p,
    & li,
    & ul,
    & ol {
        margin: 0;
        font: inherit;
        color: inherit;
    }

    & h3 {
      font: var(--font-body);
      color: var(--black);
      margin: var(--size-7) 0 var(--size-3);
    }

    & li + li,
    & :matches(p, ul, ol) + :matches(p, ul, ol) {
        margin-top: 0.8em;
    }

    & strong {
        color: var(--black);
    }

    & ol {
        margin-left: 0;
        padding-left: var(--size-7);

        & li::marker {
            font: var(--font-callout);
            font-weight: var(--bold);
            color: var(--gray-light);
        }
    }

    & ul {
        margin-left: 0;
        padding-left: 0;

        & li {
            list-style: none;
            position: relative;
            padding-left: var(--size-7);

            &::before {
                content: "";
                position: absolute;
                left: 0;
                top: .4em;
                width: var(--size-3);
                height: var(--size-3);
                background: var(--gray-light);
                border-radius: 50%;
            }
        }
    }

    @media (--to-width-1) {
        padding-left: var(--size-6);
        padding-right: var(--size-6);
    }
}
.ola_task-footer {
    padding: 0 var(--inner) var(--size-8);

    @media (--to-width-1) {
        padding-left: var(--size-6);
        padding-right: var(--size-6);
    }
}

.ola_task-info {
    padding: 0 var(--inner) var(--size-8);
    font: var(--font-caption);
    color: var(--gray);
    display: flex;
    flex-wrap: wrap;

    & > * {
        margin: 0;
        margin-right: var(--size-7);
    }

    & strong {
        color: var(--black);
    }

    @media (--to-width-1) {
        padding-left: var(--size-6);
        padding-right: var(--size-6);
    }
}



@keyframes task-open {
    from {
        opacity: 0;
        transform: translateY(-28px);
    }
    to {
        opacity: 1;
        transform: translateY(0);
    }
}
