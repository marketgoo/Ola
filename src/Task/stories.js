import React from 'react'
import { storiesOf } from '@storybook/react'

import Task from './'
import TaskBody from './Body'
import TaskFooter from './Footer'
import TaskInfo from './Info'
import TaskGroup from '../TaskGroup'
import TaskSmall from './Small'

import Button from '../Button'
import ButtonGroup from '../ButtonGroup'
import Tag from '../Tag'
import Panel from '../Panel'
import PanelContent from '../Panel/Content'

storiesOf('Task')
  .add('Error', () => (
    <Panel>
      <PanelContent title="What do you need to look at?" variant="fullwidth">
        <TaskGroup>

          <Task title="Task title demo" onToggle={e => ('Task is ' + (e.target.open ? 'open' : 'close'))}>
            <TaskInfo>
              <p>Impact: <strong>High</strong></p>
              <p>Estimated time: <strong>1-2 minutes</strong></p>
            </TaskInfo>
            <TaskBody>
              <p>Add the targeted keyword <strong>florida man</strong> to the page title to help it rank in search engine results.</p>
              <p>Here are some tips for writing an SEO-friendly title:</p>
              <ul>
                <li>Add the keyword <strong>florida man</strong> to the beginning of the title for best results.</li>
                <li>Avoid or limit STOP words, in the title tag. These words carry little value from an SEO perspective.</li>
              </ul>
              <p>Here are some tips for writing an SEO-friendly title:</p>
              <ol>
                <li>Add the keyword <strong>florida man</strong> to the beginning of the title for best results.</li>
                <li>Avoid or limit STOP words, in the title tag. These words carry little value from an SEO perspective.</li>
              </ol>
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

          <Task title="Task title demo 2" extra={<Button variant="secondary" onClick={e => {e.preventDefault();alert('click')}}>Fix now</Button>}>
            <TaskBody>
              Test content for card resolve
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

          <Task title={<>Task html title <strong>demo 3</strong></>} extra={<Button onClick={e => {e.preventDefault();alert('click')}}>Fix now</Button>}>
            <TaskBody>
              Test content for card resolve
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

        </TaskGroup>
      </PanelContent>
    </Panel>
  ))
  .add('Success', () => (
    <Panel>
      <PanelContent title="What have you done well?" variant="fullwidth">
        <TaskGroup>
          <Task title="Task title" variant="success" />
          <Task title="Task title" variant="success" extra={<Button variant="pro" onClick={e => {e.preventDefault();alert('click')}}>buy</Button>}/>
          <Task title="Task title" variant="success" extra={<Tag variant="success">Success</Tag>}/>
        </TaskGroup>
      </PanelContent>
    </Panel>
  ))
  .add('Suggested', () => (
    <Panel>
      <PanelContent title="Suggestions" variant="fullwidth">
        <TaskGroup>
          <Task title="Task title" variant="suggested" />
          <Task title="Task title" variant="suggested" />
          <Task title="Task title" variant="suggested" />
        </TaskGroup>
      </PanelContent>
    </Panel>
  ))
  .add('Reopened', () => (
    <Panel>
      <PanelContent title="Reopened" variant="fullwidth">
        <TaskGroup>
          <Task title="Task title" variant="warning">
            <TaskBody>
              Test content for card resolve
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

          <Task title="Task title" variant="warning">
            <TaskBody>
              Test content for card resolve
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

        </TaskGroup>
      </PanelContent>
    </Panel>
  ))
  .add('Mixed', () => (
    <Panel>
      <PanelContent title="Mixed" variant="fullwidth">
        <TaskGroup>

          <Task title="Task title" variant="warning">
            <TaskBody>
              <p>Text content for card resolve</p>
              <hr />
              <p>More text content for card resolve</p>
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

          <Task title="Task title" variant="suggested">
            <TaskBody>
            Text content for card resolve
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

          <Task title="Task title" variant="success" />
          <Task title="Task title" variant="suggested" />

          <Task title="Task title demo" variant="error">
            <TaskBody>
            Text content for card resolve
            </TaskBody>
            <TaskFooter>
              <ButtonGroup>
                <Button>Done! Check it now</Button>
              </ButtonGroup>
            </TaskFooter>
          </Task>

        </TaskGroup>
      </PanelContent>
    </Panel>
  ))
  .add('Small', () => (
    <Panel>
      <PanelContent>
        <TaskGroup>
          <TaskSmall>Task title</TaskSmall>
          <TaskSmall variant="suggested" href="#">Task title</TaskSmall>
          <TaskSmall variant="warning">Task title</TaskSmall>
          <TaskSmall variant="success" href="#">Task title</TaskSmall>
          <TaskSmall variant="error">Task title</TaskSmall>
        </TaskGroup>
      </PanelContent>
    </Panel>
  ))
