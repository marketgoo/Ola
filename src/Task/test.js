import React from 'react'
import Task from './'
import TaskSmall from './Small'
import Button from './../Button'
import renderer from 'react-test-renderer'

it('Default Task', () => {
  const tree = renderer
    .create(
      <Task title="Task title 1">

      </Task>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Success Task', () => {
  const tree = renderer
    .create(
      <Task title="Task title 2" variant="success">
  
      </Task>
    )
    .toJSON()
  
  expect(tree).toMatchSnapshot()
})

it('Suggested Task', () => {
  const tree = renderer
    .create(
      <Task title="Task title 3" variant="suggested">
    
      </Task>
    )
    .toJSON()
    
  expect(tree).toMatchSnapshot()
})

it('Warning Task', () => {
  const tree = renderer
    .create(
      <Task title="Task title 4" variant="warning">
      
      </Task>
    )
    .toJSON()
      
  expect(tree).toMatchSnapshot()
})

it('Error Task', () => {
  const tree = renderer
    .create(
      <Task title="Task title 5" variant="error">
        
      </Task>
    )
    .toJSON()
        
  expect(tree).toMatchSnapshot()
})

it('Task with Extra', () => {
  const tree = renderer
    .create(
      <Task 
        title="Task title 6" 
        variant="error" 
        extra={<Button onClick={() => false}>Fix now</Button>}/>
    )
    .toJSON()
          
  expect(tree).toMatchSnapshot()
})

it('Task small', () => {
  const tree = renderer
    .create(
      <TaskSmall variant="error">Task title 7</TaskSmall>
    )
    .toJSON()
          
  expect(tree).toMatchSnapshot()
})
it('Task small with link', () => {
  const tree = renderer
    .create(
      <TaskSmall variant="error" href="#">Task title 8</TaskSmall>
    )
    .toJSON()
          
  expect(tree).toMatchSnapshot()
})