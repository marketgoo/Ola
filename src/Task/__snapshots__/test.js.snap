// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Task 1`] = `
<div
  className="ola_task is-error"
>
  <div
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-medium ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2c1.63 0 3.186.317 4.666.95 1.48.634 2.756 1.486 3.827 2.557 1.07 1.071 1.923 2.347 2.557 3.827.633 1.48.95 3.035.95 4.666 0 1.63-.317 3.186-.95 4.666-.634 1.48-1.486 2.756-2.557 3.827-1.071 1.07-2.347 1.923-3.827 2.557-1.48.633-3.035.95-4.666.95-1.63 0-3.186-.317-4.666-.95-1.48-.634-2.756-1.486-3.827-2.557-1.07-1.071-1.923-2.347-2.557-3.827A11.736 11.736 0 0 1 2 14c0-1.63.317-3.186.95-4.666.634-1.48 1.486-2.756 2.557-3.827 1.071-1.07 2.347-1.923 3.827-2.557C10.814 2.317 12.369 2 14 2zm5.085 5.815c.301 0 .56.107.776.32.216.213.324.47.324.771 0 .301-.108.56-.324.776L15.543 14l4.318 4.31c.216.216.324.471.324.767 0 .3-.108.558-.324.771-.216.213-.475.32-.776.32-.3 0-.557-.105-.767-.316L14 15.542l-4.31 4.31c-.21.21-.468.316-.775.316-.301 0-.558-.105-.772-.316a1.037 1.037 0 0 1-.32-.767c0-.307.106-.565.316-.775L12.457 14 8.14 9.682a1.044 1.044 0 0 1-.315-.767c0-.301.106-.56.32-.776.213-.216.47-.324.77-.324.302 0 .56.108.776.324L14 12.457l4.318-4.318c.216-.216.472-.324.767-.324z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 1
    </span>
  </div>
</div>
`;

exports[`Error Task 1`] = `
<div
  className="ola_task is-error"
>
  <div
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-medium ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2c1.63 0 3.186.317 4.666.95 1.48.634 2.756 1.486 3.827 2.557 1.07 1.071 1.923 2.347 2.557 3.827.633 1.48.95 3.035.95 4.666 0 1.63-.317 3.186-.95 4.666-.634 1.48-1.486 2.756-2.557 3.827-1.071 1.07-2.347 1.923-3.827 2.557-1.48.633-3.035.95-4.666.95-1.63 0-3.186-.317-4.666-.95-1.48-.634-2.756-1.486-3.827-2.557-1.07-1.071-1.923-2.347-2.557-3.827A11.736 11.736 0 0 1 2 14c0-1.63.317-3.186.95-4.666.634-1.48 1.486-2.756 2.557-3.827 1.071-1.07 2.347-1.923 3.827-2.557C10.814 2.317 12.369 2 14 2zm5.085 5.815c.301 0 .56.107.776.32.216.213.324.47.324.771 0 .301-.108.56-.324.776L15.543 14l4.318 4.31c.216.216.324.471.324.767 0 .3-.108.558-.324.771-.216.213-.475.32-.776.32-.3 0-.557-.105-.767-.316L14 15.542l-4.31 4.31c-.21.21-.468.316-.775.316-.301 0-.558-.105-.772-.316a1.037 1.037 0 0 1-.32-.767c0-.307.106-.565.316-.775L12.457 14 8.14 9.682a1.044 1.044 0 0 1-.315-.767c0-.301.106-.56.32-.776.213-.216.47-.324.77-.324.302 0 .56.108.776.324L14 12.457l4.318-4.318c.216-.216.472-.324.767-.324z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 5
    </span>
  </div>
</div>
`;

exports[`Success Task 1`] = `
<div
  className="ola_task is-success"
>
  <div
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-medium ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2c1.63 0 3.186.317 4.666.95 1.48.634 2.756 1.486 3.827 2.557 1.07 1.071 1.923 2.347 2.557 3.827.633 1.48.95 3.035.95 4.666 0 1.63-.317 3.186-.95 4.666-.634 1.48-1.486 2.756-2.557 3.827-1.071 1.07-2.347 1.923-3.827 2.557-1.48.633-3.035.95-4.666.95-1.63 0-3.186-.317-4.666-.95-1.48-.634-2.756-1.486-3.827-2.557-1.07-1.071-1.923-2.347-2.557-3.827A11.736 11.736 0 0 1 2 14c0-1.63.317-3.186.95-4.666.634-1.48 1.486-2.756 2.557-3.827 1.071-1.07 2.347-1.923 3.827-2.557C10.814 2.317 12.369 2 14 2zm5.919 7c.298 0 .553.109.764.326.211.217.317.48.317.786 0 .307-.107.57-.32.79l-7.56 7.777a1.02 1.02 0 0 1-.76.321 1.03 1.03 0 0 1-.768-.321l-4.28-4.394a1.09 1.09 0 0 1-.312-.79c0-.307.106-.57.317-.787.21-.217.466-.325.764-.325s.555.11.769.33l3.51 3.611 6.79-6.994c.214-.22.47-.33.769-.33z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 2
    </span>
  </div>
</div>
`;

exports[`Suggested Task 1`] = `
<div
  className="ola_task is-suggested"
>
  <div
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-medium ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2C7.376 2 2 7.376 2 14s5.376 12 12 12 12-5.376 12-12S20.624 2 14 2zm1 19h-2v-2h2v2zm2.07-7.75l-.9.92C15.45 14.9 15 15.5 15 17h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 3
    </span>
  </div>
</div>
`;

exports[`Task small 1`] = `
<div
  className="ola_task is-small is-error"
>
  <p
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-small ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2c1.63 0 3.186.317 4.666.95 1.48.634 2.756 1.486 3.827 2.557 1.07 1.071 1.923 2.347 2.557 3.827.633 1.48.95 3.035.95 4.666 0 1.63-.317 3.186-.95 4.666-.634 1.48-1.486 2.756-2.557 3.827-1.071 1.07-2.347 1.923-3.827 2.557-1.48.633-3.035.95-4.666.95-1.63 0-3.186-.317-4.666-.95-1.48-.634-2.756-1.486-3.827-2.557-1.07-1.071-1.923-2.347-2.557-3.827A11.736 11.736 0 0 1 2 14c0-1.63.317-3.186.95-4.666.634-1.48 1.486-2.756 2.557-3.827 1.071-1.07 2.347-1.923 3.827-2.557C10.814 2.317 12.369 2 14 2zm5.085 5.815c.301 0 .56.107.776.32.216.213.324.47.324.771 0 .301-.108.56-.324.776L15.543 14l4.318 4.31c.216.216.324.471.324.767 0 .3-.108.558-.324.771-.216.213-.475.32-.776.32-.3 0-.557-.105-.767-.316L14 15.542l-4.31 4.31c-.21.21-.468.316-.775.316-.301 0-.558-.105-.772-.316a1.037 1.037 0 0 1-.32-.767c0-.307.106-.565.316-.775L12.457 14 8.14 9.682a1.044 1.044 0 0 1-.315-.767c0-.301.106-.56.32-.776.213-.216.47-.324.77-.324.302 0 .56.108.776.324L14 12.457l4.318-4.318c.216-.216.472-.324.767-.324z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 7
    </span>
  </p>
</div>
`;

exports[`Task small with link 1`] = `
<a
  className="ola_task is-small is-error"
  href="#"
>
  <p
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-small ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2c1.63 0 3.186.317 4.666.95 1.48.634 2.756 1.486 3.827 2.557 1.07 1.071 1.923 2.347 2.557 3.827.633 1.48.95 3.035.95 4.666 0 1.63-.317 3.186-.95 4.666-.634 1.48-1.486 2.756-2.557 3.827-1.071 1.07-2.347 1.923-3.827 2.557-1.48.633-3.035.95-4.666.95-1.63 0-3.186-.317-4.666-.95-1.48-.634-2.756-1.486-3.827-2.557-1.07-1.071-1.923-2.347-2.557-3.827A11.736 11.736 0 0 1 2 14c0-1.63.317-3.186.95-4.666.634-1.48 1.486-2.756 2.557-3.827 1.071-1.07 2.347-1.923 3.827-2.557C10.814 2.317 12.369 2 14 2zm5.085 5.815c.301 0 .56.107.776.32.216.213.324.47.324.771 0 .301-.108.56-.324.776L15.543 14l4.318 4.31c.216.216.324.471.324.767 0 .3-.108.558-.324.771-.216.213-.475.32-.776.32-.3 0-.557-.105-.767-.316L14 15.542l-4.31 4.31c-.21.21-.468.316-.775.316-.301 0-.558-.105-.772-.316a1.037 1.037 0 0 1-.32-.767c0-.307.106-.565.316-.775L12.457 14 8.14 9.682a1.044 1.044 0 0 1-.315-.767c0-.301.106-.56.32-.776.213-.216.47-.324.77-.324.302 0 .56.108.776.324L14 12.457l4.318-4.318c.216-.216.472-.324.767-.324z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 8
    </span>
  </p>
</a>
`;

exports[`Task with Extra 1`] = `
<div
  className="ola_task is-error"
>
  <div
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-medium ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2c1.63 0 3.186.317 4.666.95 1.48.634 2.756 1.486 3.827 2.557 1.07 1.071 1.923 2.347 2.557 3.827.633 1.48.95 3.035.95 4.666 0 1.63-.317 3.186-.95 4.666-.634 1.48-1.486 2.756-2.557 3.827-1.071 1.07-2.347 1.923-3.827 2.557-1.48.633-3.035.95-4.666.95-1.63 0-3.186-.317-4.666-.95-1.48-.634-2.756-1.486-3.827-2.557-1.07-1.071-1.923-2.347-2.557-3.827A11.736 11.736 0 0 1 2 14c0-1.63.317-3.186.95-4.666.634-1.48 1.486-2.756 2.557-3.827 1.071-1.07 2.347-1.923 3.827-2.557C10.814 2.317 12.369 2 14 2zm5.085 5.815c.301 0 .56.107.776.32.216.213.324.47.324.771 0 .301-.108.56-.324.776L15.543 14l4.318 4.31c.216.216.324.471.324.767 0 .3-.108.558-.324.771-.216.213-.475.32-.776.32-.3 0-.557-.105-.767-.316L14 15.542l-4.31 4.31c-.21.21-.468.316-.775.316-.301 0-.558-.105-.772-.316a1.037 1.037 0 0 1-.32-.767c0-.307.106-.565.316-.775L12.457 14 8.14 9.682a1.044 1.044 0 0 1-.315-.767c0-.301.106-.56.32-.776.213-.216.47-.324.77-.324.302 0 .56.108.776.324L14 12.457l4.318-4.318c.216-.216.472-.324.767-.324z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 6
    </span>
    <div
      className="ola_task-extra"
    >
      <button
        className="ola_button"
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="ola_button-text"
        >
          Fix now
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Warning Task 1`] = `
<div
  className="ola_task is-warning"
>
  <div
    className="ola_task-header"
  >
    <svg
      className="ola_icon is-medium ola_task-status"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M25.89 23.74L14.755 4.431A.87.87 0 0 0 14 4c-.312 0-.6.164-.755.431L2.112 23.74a.828.828 0 0 0 .006.841.87.87 0 0 0 .747.419h22.27c.304 0 .59-.16.747-.419a.834.834 0 0 0 .008-.841zM15 22h-2v-2h2v2zm0-4h-2v-8h2v8z"
      />
    </svg>
    <span
      className="ola_task-title"
    >
      Task title 4
    </span>
  </div>
</div>
`;
